docs/cryptodoc/src/05_10_cmce.rst

```diff
@@ -0,0 +1,469 @@
+To represent GF elements, we utilize the class ``Classic_McEliece_GF``. Each
+element is defined by two polynomials: one defines its value, and the other
+defines the modulus of the field. These values are represented as unsigned
+integers. For example, the element :math:`z^3+z+1` is represented by the unsigned
+integer ``0b1011``.
+
+The GF elements support various operations such as addition, multiplication,
+and inversion. These operations are implemented in constant time for operands
+within the same field. Multiplication is performed using a constant-time
+schoolbook multiplication algorithm with a consecutive constant-time reduction.
+Inversion is achieved using Fermat's little theorem: :math:`a^{-1} = a^{q-2}`.
+For that, exponentiation is implemented using a simple square-and-multiply
+algorithm. It is worth noting that square-and-multiply has no timing
+side-channel vulnerabilities since the exponent remains constant for each field.
+
+.. _pubkey/cmce/field_ordering:
+
+Field Ordering
+^^^^^^^^^^^^^^
+
+Section 8.2 of [CMCE-ISO]_ defines the field ordering as a permutation
+of elements in the Galois field :math:`\mathbb{F}_{q}`. This ordering is
+required for generating a random sequence :math:`\alpha_0,...,\alpha_{n-1}` of distinct
+:math:`\mathbb{F}_{q}` elements, which is necessary for creating the Goppa
+code. Botan's ``Classic_McEliece_Field_Ordering`` serves as a container for the
+field ordering and includes the algorithm to generate it. To sort the pairs
+:math:`(a_i, i)` (as described in Step 3 of Section 8.2 in [CMCE-ISO]_), a
+constant-time bitonic sort implementation is utilized. This sorting algorithm is
+particularly suitable for sets with a power of two elements.
+
+Another vital role of the field ordering class is managing the Beneš network.
+As Section 9.2.10 of [CMCE-ISO]_ outlines, the Beneš network stores the field
+ordering as control bits in a compact form. Botan employs
+a constant-time implementation of the ``controlbits`` algorithm presented in
+Fig. 7.1 of [CBits]_ to create control bits, and the ``permutation`` algorithm
+to reconstruct the field ordering from the control bits. The creation
+algorithm also utilizes constant-time bitonic sorting.
+
+
+.. _pubkey/cmce/poly:
+
+Polynomial Arithmetic
+^^^^^^^^^^^^^^^^^^^^^
+
+The Classic McEliece algorithm uses elements in the polynomial ring
+:math:`\mathbb{F}_q [y]` in multiple places. The monic irreducible
+polynomial defines the Goppa code, and performing decapsulation requires an
+error locator polynomial. Botan's ``Classic_McEliece_Poly`` class is designed
+to represent these polynomials. It consists of a vector of GF elements, which
+serve as the polynomial coefficients. Additionally, this class provides
+the necessary logic to evaluate the polynomial at a given point in
+:math:`\mathbb{F}_q`.
+
+Polynomial arithmetic is required to obtain the monic irreducible polynomial
```

Comment:
Maybe make it explicit that the input to this algorithm is a seed (I stumbled upon this in the keygen and did at first not make a connection to this section)